compiler/rustc_mir_build/src/thir/cx/expr.rs

```diff
@@ -863,15 +859,7 @@ fn convert_path_expr<'a, 'tcx>(
         Res::Def(DefKind::Static, id) => {
             let ty = cx.tcx.static_ptr_ty(id);
             let temp_lifetime = cx.region_scope_tree.temporary_scope(expr.hir_id.local_id);
-            let kind = if cx.tcx.is_thread_local_static(id) {
-                ExprKind::ThreadLocalRef(id)
-            } else {
-                let ptr = cx.tcx.create_static_alloc(id);
-                ExprKind::StaticRef {
-                    literal: ty::Const::from_scalar(cx.tcx, Scalar::Ptr(ptr.into()), ty),
-                    def_id: id,
-                }
-            };
+            let kind = convert_static_ref(cx, id, ty);
```

Comment:
"kind" is a really confusing variable name here, this is an expression after all.